### ISSUE TITLE
Model Binding for ASP.net MVC 5 controller?

### ISSUE THREAD

**Issue by colinwurtz** [timheuer/alexa-skills-dotnet#83]
I have an ASP.Net MVC 5 application with a controller method I'm using to handle incoming Alexa requests. Thought this library looked promising to automatically bind to the request but I'm not having any luck. Am I doing something wrong here or trying to use this for something it wasn't made for?

```
using Alexa.NET.Request;

     [HttpPost]
        [AllowAnonymous]
        public dynamic Alexa(SkillRequest input)
        {

            Debug.Write(input.Request.Type);

            return JsonConvert.SerializeObject(new {Data="blah"});
        }

```
When I hit the endpoint with standard Json from Alexa I can't parse the input and either get `null` for input or errors like the one below.. Thanks for the help.
```

<!DOCTYPE html>
<html>
    <head>
        <title>Cannot create an abstract class.</title>
        <meta name="viewport" content="width=device-width" />
        <style>
         body {font-family:"Verdana";font-weight:normal;font-size: .7em;color:black;} 
         p {font-family:"Verdana";font-weight:normal;color:black;margin-top: -5px}
```

**Comment by timheuer** [comment]
Hi @colinwurtz - it's hard to tell what you may be hitting.  Your Alexa app will be calling a specific endpoint and you need to make sure you are specifically providing that endpoint URI that Alexa will POST to.  I'm not sure if that is what you may be running in to?  The fact that sometimes you are getting null and sometimes getting the ```Cannot create an abstract class``` tells me at least it is getting to your code though.

Have you tried outside of the controller model you have just to isolate that you are indeed getting a payload correctly to your API?

**Comment by stoiveyp** [comment]
A Create abstract class error makes me think it's receiving the JSON but not able to deserialize it because it's not using JSON.NET conversion on the request type.

There's this article about the issue on SO -  https://stackoverflow.com/questions/23995210/how-to-use-json-net-for-json-modelbinding-in-an-mvc5-project

**Comment by colinwurtz** [comment]
I tried several different implementations of the JSON.net model binder mentioned on the SO post but still kept hitting the "abstract class error". 

It seems the way the SkillRequest class is set up just doesn't play very nice with ASP.Net MVC which is fine. 

I ended up using this model mapping of the skill request and the data came through. Thanks for your help on this @stoiveyp .

https://github.com/Quorralyne/LCARS-Alexa-Skill-WebAPI/blob/a4beff4a737a70ecbcb41bebd21cf92db06e5121/LCARSAlexaSkill/Models/AlexaRequest.cs
